code/math/vecmat.cpp

```diff
@@ -2273,13 +2278,12 @@ void vm_vec_random_in_sphere(vec3d *out, const vec3d *in, float radius, bool on_
 {
 	vec3d temp;
 
-	// Uniformly distributing each coordinate of a vector then normalizing results in a uniform sphere distribution
-	util::UniformFloatRange coords(-1.0f,1.0f);
-	vm_vec_make( &temp, coords.next(), coords.next(), coords.next() );
-	vm_vec_normalize(&temp);
+	float z = util::UniformFloatRange(-1, 1).next(); // Take a 2-sphere slice
+	float phi = util::UniformFloatRange(0.0f, PI2).next();
+	vm_vec_make(&temp, sqrtf(1.0f - z * z) * cosf(phi), sqrtf(1.0f - z * z) * sinf(phi), z); // Using the z-vec as the starting point
 
-	// We then add the scaled result to the initial position to get the final position
-	vm_vec_scale_add(out, in, &temp, on_surface ? radius : util::UniformFloatRange(0.0f, radius).next());
+	// cube root because scaling inward increases the probability desnity by the cube of its distance towards the center
+	vm_vec_scale_add(out, in, &temp, on_surface ? radius : powf(util::UniformFloatRange(0.0f, 1.0f).next(), 0.333f) * radius);
```

Comment:
It would be nice to have a `bias_towards_center` parameter just like the companion function